esmvaltool/diag_scripts/monitor/multi_datasets.py

```diff
@@ -1360,14 +1440,115 @@ def create_profile_plot(self, datasets, short_name):
                 for netcdf_path in netcdf_paths:
                     provenance_logger.log(netcdf_path, provenance_record)
 
+    def create_1d_profile_plot(self, datasets, short_name):
+        """Create 1D profile plot."""
+        plot_type = '1d_profile'
+        if plot_type not in self.plots:
+            return
+
+        if not datasets:
+            raise ValueError(f"No input data to plot '{plot_type}' given")
+
+        logger.info("Plotting %s", plot_type)
+        fig = plt.figure(**self.cfg['figure_kwargs'])
+        axes = fig.add_subplot()
+
+        multi_dataset_facets = self._get_multi_dataset_facets(datasets)
+
+        # Plot all datsets in one single figure
+        ancestors = []
+        cubes = {}
+        for dataset in datasets:
+            ancestors.append(dataset['filename'])
+            cube = dataset['cube']
+            cubes[self._get_label(dataset)] = cube
+            self._check_cube_dimensions(cube, plot_type)
+
+            # Plot 1D profile
+            plot_kwargs = self._get_plot_kwargs(plot_type, dataset)
+            plot_kwargs['axes'] = axes
+
+            iris.plot.plot(cube, **plot_kwargs)
+
+        # Default plot appearance
+        axes.set_title(multi_dataset_facets['long_name'])
+        axes.set_xlabel(f"{short_name} [{multi_dataset_facets['units']}]")
+        z_coord = cube.coord(axis='Z')
+        axes.set_ylabel(f'{z_coord.long_name} [{z_coord.units}]')
+
+        # apply logarithmic axes
+        if self.plots[plot_type]['log_y']:
+            axes.set_yscale('log')
+            axes.get_yaxis().set_major_formatter(
+                FormatStrFormatter('%.1f'))
+        if self.plots[plot_type]['show_y_minor_ticklabels']:
+            axes.get_yaxis().set_minor_formatter(
+                FormatStrFormatter('%.1f'))
+        else:
+            axes.get_yaxis().set_minor_formatter(NullFormatter())
+        if self.plots[plot_type]['log_x']:
+            axes.set_xscale('log')
+            # major and minor ticks
+            x_major = LogLocator(base=10.0, numticks=12)
+            axes.get_xaxis().set_major_locator(x_major)
+            x_minor = LogLocator(base=10.0,
+                                 subs=np.arange(1.0, 10.0) * 0.1,
+                                 numticks=12)
+            axes.get_xaxis().set_minor_locator(x_minor)
+            axes.get_xaxis().set_minor_formatter(NullFormatter())
```

Comment:
It's probably better to not hardcode the number of ticks here, you should be able to specify the tick locations in the recipe with

```yaml
pyplot_kwargs:
  xticks: [1, 2, 3]
```

(not tested though)

Reply:
I actually had my issues here with the minor Log ticks. This seems an matplotlib issue (see [here](https://stackoverflow.com/questions/44078409/matplotlib-semi-log-plot-minor-tick-marks-are-gone-when-range-is-large) Would you be fine when I add numticks as an option with default 12. Those restrained minor ticks are only applied in the xlog case and can still be specified in the recipe for the non-xlog plots.

Reply:
numticks is actually not changing the number of ticks, but need to be set equal or larger the number of displayed major ticks.